src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelThread.cs

```diff
@@ -18,6 +19,8 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Internal
     /// </summary>
     public class KestrelThread
     {
+        public const int MaxPooledWriteReqs = 1024;
```

Comment:
You could unconst this and set it to zero before disposing all the `UvWriteReq`s in `Stop`. This will prevent new `UvWriteReq`s from potentially being added to the pool afterwards and finalized.